### PR TITLE
Hide AKS, EKS, and GKE cards when their kontainer drivers are deactivated

### LINKED ISSUE #12898: Disable new hosted provider extensions
- Some customers would like to remove the ability to provision to hosted providers
- Previously this was possible by removing kontainer drivers
- We now load extensions that use the kontainer drivers but don't check if they exist or are active first
  - extensions are collected  https://github.com/rancher/dashboard/blob/master/shell/edit/provisioning.cattle.io.cluster/index.vue#L154
  - extensions are added as subTypes https://github.com/rancher/dashboard/blob/master/shell/edit/provisioning.cattle.io.cluster/index.vue#L369
- We should also consider when extensions are associated with missing or disabled node drivers

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
### Summary
Fixes #12898 

### Occurred changes and/or fixed issues
This PR updates the AKS, EKS, and GKE ui extensions to hide their respective cluster creation cards when their kontainer driver is deactivated.

### Technical notes summary
The PR changes the behavior of the `disabled` property of the ClusterProvisioner class so that it aligns with what happens when a node/kontainer driver with custom ui is deactivated. Rather than showing disabled cards the cards are gone entirely.

### Areas or cases that should be tested
- disable aks, eks, gke kontainer drivers and verify that the associated card is gone from the cluster creation page after each is disabled
- re-enable those providers and verify that the cards are back
- verify that clicking the cards loads the newer vue ui 

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/core/types-provisioning.ts
Comment (richard-cox) [inline shell/core/types-provisioning.ts:105]: I don't think we should switch the functionality of an existing function.

We have a couple of different options
- Get providers to declare they are a kontainer driver and the kontainer driver name. We would then do the check dashboard side. JSDOC / dev kit docs / release notes should make it clear that we won't show the provider unless we find a matching kontainer with the driver name.
  - We could chose to make this optional, so only when supplied and we don't find a matching kontainer driver do we hide the option 
- Add a specific `hide` provider prop

Comment (mantis-toboggan-md) [reply shell/core/types-provisioning.ts:105]: >Add a specific hide provider prop

I think this is more likely to be useful to other extension developers than adding something so kontainer-specific, so I'll try this approach

File: shell/edit/provisioning.cattle.io.cluster/index.vue
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/index.vue:453]: Moving the filtering to this step (so that 'hidden' provisioners are still in the `subtypes` array) ensures that if a user creates a cluster using one of these provisioners then disables the associated driver, the right ui still loads when editing that cluster. Not sure if it really makes sense for a user to get themselves into that scenario, but it feels like we should load the right UI if they do anyway

File: cypress/e2e/tests/pages/manager/cluster-manager.spec.ts
Comment (richard-cox) [inline cypress/e2e/tests/pages/manager/cluster-manager.spec.ts:93]: To ensure customer use case is covered we should either (ideally) have a test that entirely removes the the kontainer drivers or request manual QA for them. We could be sneaky with the former and just intercept the kontainer request and return an empty list.

Sorry should have thought of this yesterday

File: shell/core/types-provisioning.ts
Comment (richard-cox) [inline shell/core/types-provisioning.ts:112]: Not quite sure how we do changes to extension api changes. @aalves08 + @jordojordo , should this change include a bump to the shell package version indicating that a new release will be required at some point... or do we have another notification process?

This doesn't block the PR or customer (right now), but will when they upgrade / use 2.9.6

File: shell/edit/provisioning.cattle.io.cluster/index.vue
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/index.vue:460]: When all hosted / cluster providers are disabled there's a nit picky vertical space issue

image

The gap between line and `Provision new nodes and...` isn't there and isn't the same as above and below `Use existing nodes`. It's not been introduced in this PR but will be more noticeable without it

**Review by richard-cox — DISMISSED** [review]
I've left three comments to be resolved, though none are blocking this from merging and can be tracked in issues for later.

let me know if you'd like me to do / split between us some of the nit picky ones!

File: shell/edit/provisioning.cattle.io.cluster/index.vue
Comment (mantis-toboggan-md) [reply shell/edit/provisioning.cattle.io.cluster/index.vue:460]: updated to make the spacing consistent across all 3 sections:
Screenshot 2024-12-19 at 11 57 26 AM
Screenshot 2024-12-19 at 11 57 10 AM

File: cypress/e2e/tests/pages/manager/cluster-manager.spec.ts
Comment (richard-cox) [reply cypress/e2e/tests/pages/manager/cluster-manager.spec.ts:93]: That worked fine for me. I did push a change with it and some tweaks
- prefixed the intercept url with `/`. don't think it's needed, but good to have
- added a check to ensure the groups are correct (no cluster drivers === no top group). i didn't noticed a couple of weird things whilst doing so
  - the cypress screenshot / live browser acts incorrectly, sometimes this would show the hidden row. guessing this is somehow related to the rendering then not using the tests intercepted api request
  - when listing group titles `first` and `eq` behaved weirdly given the live browser

**Review by richard-cox — APPROVED** [review]
Ok, https://github.com/rancher/dashboard/pull/12925 should fix the failing storybook gate. Once that merges this just needs a rebase and can go in.

@mantis-toboggan-md Given backports might also be affected I'd be ok with cherry-picking the commit from that PR and included it in the backport PRs. Does that sound ok your side?

File: shell/core/types-provisioning.ts
Comment (richard-cox) [reply shell/core/types-provisioning.ts:112]: Asked in slack, but also creted https://github.com/rancher/dashboard/issues/12930 for docs

Comment (aalves08) [reply shell/core/types-provisioning.ts:112]: It should include a bump in the shell package + the release of that shell version via a tagged release on our repository. There's an internal doc that I can refer to.

**Pull request merged**
